### ISSUE TITLE
Information on the total options count is missing in the screen reader announcement after filtering

### ISSUE THREAD

**Issue by andriibuts** [JedWatson/react-select#5496]
Repro link: https://codesandbox.io/s/react-select-v5-sandbox-aria-context-7bcp8l?file=/example.js

This issue is a follow-up of https://github.com/JedWatson/react-select/issues/5083

1. There are 10 options in total (example number).
2. A search phrase is entered (e.g., "o"), the dropdown is filtered, and only 4 options are available now.

SR will read:

```
option Ocean selected, 1 of 4. 4 results available for search term o
```

The numbers correctly reflect post-filtering results, but the context about the total number of options is missing.

Could we update the test to something like this:
```
option Ocean selected, 1 of 4. 4 results available out of 10 for search term o...
```
Could we add `out of {totalOptions}`  after the `results available`?

On a separate note, I can create an MR with the requested feature.
